tests/SimplifyTest.elm

```diff
@@ -16434,6 +16434,7 @@ dictSizeTests =
 a = Dict.size
 a = Dict.size b
 a = Dict.size (Dict.fromList [b, c, d])
+a = Dict.size (Dict.fromList [((),1), ((),2), ((),3)])
```

Comment:
I'm super confused as to why these last two are not reported. Are they not the same as the tests you've added below?

EDIT: Aaah, gotcha, it's because the number can be wrong if some of the entries have the same key. Maybe edit the title of the new tests to make that explicit, because it was quite confusing to me.
Also, we *should* be simplifying the line I just commented on to `1`. Maybe to be done later, but I don't think it should be part of the "okay" tests.

Reply:
units aren't comparable.
```
> () < ()
-- TYPE MISMATCH ---------------------------------------------------------- REPL

I cannot do a comparison with this value:

3|   () < ()
     ^^
The left side of (<) is a unit value:

    ()

But (<) only works on Int, Float, Char, and String values. It can work on lists
and tuples of comparable values as well, but it is usually better to find a
different path.

Hint: I only know how to compare ints, floats, chars, strings, lists of
comparable values, and tuples of comparable values.
```

Reply:
Oh yeah they aren't. But that also means that technically `Dict.fromList [ ( (), 1 ) ]` doesn't compile :smile:

Reply:
very true as well, removed those from the tests